File: task3/javascript/Sub-Task3.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('Enter the number of rows: ', (n) => {
    n = parseInt(n);
    console.log("Pattern 1");

    for (let i = 1; i <= (n + 1) / 2; i++) {
        console.log(" ".repeat((n + 1) / 2 - i) + "*".repeat((i * 2) - 1));
    }

    for (let i = (n + 1) / 2 + 1; i <= n; i++) {
        console.log(" ".repeat(i - (n + 1) / 2) + "*".repeat((n + 1 - i) * 2 - 1));
    }

    rl.close();
});

File: task3/rust/Sub-Task3.rs
fn print_pattern(n: usize) {
    println!("Pattern 1");

    for i in 1..=((n + 1) / 2) {
        let spaces = (n + 1) / 2 - i;
        let stars = (i * 2) - 1;
        println!("{: >width$}{}", "", "*".repeat(stars), width = spaces);
    }

    for i in ((n + 1) / 2 + 1)..=n {
        let spaces = i - (n + 1) / 2;
        let stars = (n + 1 - i) * 2 - 1;
        println!("{: >width$}{}", "", "*".repeat(stars), width = spaces);
    }
}

fn main() {
    let mut input = String::new();
    println!("Enter the number of rows:");
    std::io::stdin().read_line(&mut input).unwrap();
    let n: usize = input.trim().parse().unwrap();

    print_pattern(n);
}
